### ISSUE TITLE
Error: Debug Failure. Expected declaration to have at least one class-like declaration in 'getConstructSignatureDefinition' during 'definition'

### ISSUE THREAD

**Issue by aozgaa** [Microsoft/TypeScript#21785]
This issue comes from crash dumps in telemetry. We've tried to de-duplicate issues on a best-effort basis, comparing the sequence of methods called and the command requested while ignoring line numbers.
**TypeScript Version Prefix**: 2.7.1
**VSCode Version** 1.20.0-insider
**command requested**: definition
**hitting sessions**: 5528
**proportion of all sessions**: 0.02063664497728401
**stack**
```
Error: Debug Failure. Expected declaration to have at least one class-like declaration
    at getConstructSignatureDefinition (tsserver.js:70840:88)
    at getDefinitionFromSymbol (tsserver.js:70837:20)
    at Object.getDefinitionAtPosition (tsserver.js:70767:20)
    at Object.getDefinitionAtPosition (tsserver.js:82909:38)
    at IOSession.Session.getDefinition (tsserver.js:89386:64)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (tsserver.js:88894:61)
    at unknown (tsserver.js:90355:88)
    at IOSession.Session.executeWithRequestId (tsserver.js:90346:28)
    at IOSession.Session.executeCommand (tsserver.js:90355:33)
    at IOSession.Session.onMessage (tsserver.js:90375:35)
    at Interface.<anonymous> (tsserver.js:91576:27)
    at emitOne (events.js:96:13)
    at Interface.emit (events.js:191:7)
    at Interface._onLine (readline.js:241:10)
    at Interface._normalWrite (readline.js:384:12)
    at Socket.ondata (readline.js:101:10)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:191:7)
    at readableAddChunk (_stream_readable.js:178:18)
    at Socket.Readable.push (_stream_readable.js:136:10)
    at Pipe.onread (net.js:560:20)
```

**Comment by mhegazy** [comment]
looks like a duplicate of https://github.com/Microsoft/TypeScript/issues/21501. should be fixed by  #21740

**Comment by aozgaa** [comment]
This issue reproduced in the following context:
**TypeScript Version Prefix**: 2.7.2
**VSCode Version** 1.20.0
**command requested**: definition
**hitting sessions**: 152
**proportion of all sessions**: 0.035381750465549346
**stack**
```
Error: Debug Failure. Expected declaration to have at least one class-like declaration
    at getConstructSignatureDefinition (tsserver.js:71038:88)
    at getDefinitionFromSymbol (tsserver.js:71035:20)
    at Object.getDefinitionAtPosition (tsserver.js:70965:20)
    at Object.getDefinitionAtPosition (tsserver.js:83107:38)
    at IOSession.Session.getDefinition (tsserver.js:89584:64)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (tsserver.js:89092:61)
    at unknown (tsserver.js:90553:88)
    at IOSession.Session.executeWithRequestId (tsserver.js:90544:28)
    at IOSession.Session.executeCommand (tsserver.js:90553:33)
    at IOSession.Session.onMessage (tsserver.js:90573:35)
    at Interface.<anonymous> (tsserver.js:91774:27)
    at emitOne (events.js:96:13)
    at Interface.emit (events.js:191:7)
    at Interface._onLine (readline.js:241:10)
    at Interface._normalWrite (readline.js:384:12)
    at Socket.ondata (readline.js:101:10)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:191:7)
    at readableAddChunk (_stream_readable.js:178:18)
    at Socket.Readable.push (_stream_readable.js:136:10)
    at Pipe.onread (net.js:560:20)
```
